src/cmd/flux-pstree.py

```diff
@@ -0,0 +1,366 @@
+        setattr(namespace, "filtered", True)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="flux-pstree", formatter_class=flux.util.help_formatter()
+    )
+    parser.add_argument(
+        "-a",
+        "--all",
+        action=FilterTrueAction,
+        help="Include all jobs for current user in all states",
+    )
+    parser.add_argument(
+        "-c",
+        "--count",
+        action=FilterAction,
+        type=int,
+        metavar="N",
+        default=1000,
+        help="Limit to N jobs per instance (default 1000)",
+    )
+    parser.add_argument(
+        "-f",
+        "--filter",
+        action=FilterAction,
+        type=str,
+        metavar="STATE|RESULT",
+        default="running",
+        help="list jobs with specific job state or result",
+    )
+    parser.add_argument(
+        "-x",
+        "--extended",
+        action="store_true",
+        help="print extended details before tree output",
+    )
+    parser.add_argument(
+        "-l",
+        "--long",
+        action="store_false",
+        dest="truncate",
+        help="do not truncate long lines to terminal width",
+    )
+    parser.add_argument(
+        "-L",
+        "--level",
+        type=int,
+        metavar="N",
+        default=999,
+        help="only descend N levels",
+    )
+    parser.add_argument(
+        "-p",
+        "--parent-ids",
+        action="store_true",
+        help="include jobids in parent labels",
+    )
+    parser.add_argument(
+        "-n",
+        "--no-header",
+        action="store_true",
+        help="Suppress header with -x, --extended",
+    )
+    parser.add_argument(
+        "-X",
+        "--no-combine",
+        action="store_false",
+        dest="combine_children",
+        help="disable combination of identical children",
+    )
+    parser.add_argument(
+        "-o",
+        "--label",
+        metavar="FORMAT",
+        help="change label format (default='{name}')",
+    )
+    parser.add_argument(
+        "--parent-label",
+        metavar="FORMAT",
+        help="change label format for parent only",
+    )
+    parser.add_argument(
+        "--detail-format",
+        metavar="FORMAT",
+        help="specify output format for extended details " + "(implies -x, --extended)",
+        dest="prefix_format",
+    )
+    parser.add_argument(
+        "-d",
+        "--details",
+        metavar="NAME",
+        help="Select a named extended details format ("
+        + ",".join(DETAILS_FORMAT.keys())
+        + ")",
+    )
+    parser.add_argument(
+        "-C",
+        "--compact",
+        action="store_true",
+        help="Use compact tree connectors",
+    )
+    parser.add_argument(
+        "--ascii",
+        action="store_true",
+        help="Use ASCII character tree connectors",
+    )
+    parser.add_argument(
+        "--no-skip-root",
+        action="store_false",
+        dest="skip_root",
+        help="Include root in output",
+    )
```

Comment:
very first time I ran this I got:

```
flux
flux
```

and I thought something was broken.  But running `flux pstree --no-skip-root`, I got what I thought I would see

```
.
├── flux
└── flux
```

dunno if default should be with root?

Reply:
I struggled with this as well, and had it the way you suggest for awhile. The problem I ran into is that including root doesn't add much useful data, except when you catch a couple instances before they have run any jobs as you have here, and makes it more likely the tree will overflow the terminal.

However, maybe my contrived examples were deeper than in the common case, so possibly it won't be a problem in reality.

Another idea would be to include the "root" in the normal case, but exclude in when printing "detailed" information, in which case the output would be less likely to be confusing and we'd be more likely to exceed the current terminal width.

Reply:
> Another idea would be to include the "root" in the normal case, but exclude in when printing "detailed" information, in which case the output would be less likely to be confusing and we'd be more likely to exceed the current terminal width.

I like this idea.

In time perhaps a jobid input would go along with `flux pstree`?  In that case I could see `skip-root` being normal (sort of like when you do `pstree PID`).  But for the general listing of all jobs of a user, listing root seems right?  Sort of like the default output of `pstree`?

Reply:
`flux pstree` can already take a list of JOBIDs.

The real problem with "root" is that we can't easily create a `JobInfo` object for it, and for test instances or root instances run under a foreign RM there isn't even a jobid, so the comparison to `pstree` falls flat there. However, since the root is handled specially anyway and the label is always forced to `.`, maybe all that is fine.

I can also try to make a "fake" JobInfo object for the enclosing instance and see how that goes.